src/views/labs/LabsExplore/LabsExplore.vue

```diff
@@ -30,34 +32,35 @@
       DropdownSidebarPanel,
     },
   })
-  export default class LabsExploreView extends Vue {
-    data: LabsExploreData | null = null;
+  export default class LabsExploreView extends Mixins(PageDataMixin) {
+    dynamicData: LabsExploreData | null = null;
 
-    static async query(route: Route) {
+    get labsData(): LabsExploreData {
+      return this.dynamicData ?? this.pageData as LabsExploreData;
+    }
+
+    async query(route: Route) {
       const { sort } = route.query;
-      return (await axios.get('/get/?type=get_labs_for_lab_cards&size=18&skip=0', {
+      return (await this.$http.get('/get/?type=get_labs_for_lab_cards&size=18&skip=0', {
         params: {
           order: route.query.sort,
           filters: route.query.filters && (route.query.filters as string).split(','),
         },
       })).data.data as LabsExploreData;
     }
 
-    async beforeRouteEnter(to: Route, from: Route, next: RouteCallback<LabsExploreView>) {
-      const data = await LabsExploreView.query(to);
-      next((vm) => {
-        vm.data = data;
-      });
+    async fetchPageData(): Promise<any> {
+      return this.query(this.$route);
     }
 
     async beforeRouteUpdate(to: Route, from: Route, next: RouteCallback<LabsExploreView>) {
-      const data = await LabsExploreView.query(to);
-      this.data = data;
+      const data = await this.query(to);
+      this.dynamicData = data;
       next();
     }
 
-    private filters = [new Filter('active', 'Active'), new Filter('inactive', 'Inactive')];
+    private filters: Filter[] = [{ value: 'active', text: 'Active' }, { value: 'inactive', text: 'Inactive' }];
```

Comment:
Note to self - check if this has automplete

Reply:
Interfaces do indeed have autocomplete!